content/blog/2022-06-07-using-gcp-remotes-in-dvc.md

```diff
@@ -0,0 +1,178 @@
+  so we're going to go through a tutorial for doing this with GCP.
+picture: 2022-06-07/dvc-gcp.png
+pictureComment: Using GCP Remotes in DVC
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/preventing-stale-models-in-production/1137
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - DVC Remotes
+  - Collaboration
+---
+
+When you’re working on a data science project that has huge datasets, it’s
+common to store them in cloud storage. You’ll also be working with different
+versions of the same datasets to train a model, so it’s crucial to have a tool
+that enables you to do this quickly and easily. That’s why we’re going to do a
+quick walkthrough of how to set up a remote in a GCP storage bucket and handle
+data versioning with [DVC](https://dvc.org/doc).
+
+We’ll start by creating a new storage bucket in our GCP account, then we’ll show
+how you can add DVC to your project, and finally, we’ll make updates to the
+dataset with DVC commands. We’ll be working with
+[this repo](https://github.com/iterative/stale-model-example) if you want an
+example to play with. By the time you finish, you should be able to create this
+setup for any machine learning project using an GCP remote.
+
+## Set up a GCP storage bucket
+
+Make sure that you already have a
+[GCP account](https://console.cloud.google.com). You’ll need a valid credit card
+to create a new account. Once you’re logged into your account, you should see a
+screen like this with some of the services GCP offers.
+
+![GCP initial page](/uploads/images/2022-06-07/gcp_initial_page.png)
+
+From here, you'll need to create a new project. Search for "create a project"
+and click the "IAM & Admin" option. You'll enter the name of the project, which
+is `Bicycle Project`, and choose the organization and location and click the
+`Create` button. This will take you to your project dashboard and show you all
+of the stats and settings you have available.
+
+![create a new GCP project](/uploads/images/2022-06-07/gcp_new_project.png)
+
+Then you need to go to `Cloud Storage` in the left sidebar to create a bucket to
+store the data. When you get to the Cloud Storage page, you should see something
+similar to this and you’ll click the `Create Bucket` button.
+
+![create_gcp_bucket.png](/uploads/images/2022-06-07/create_gcp_bucket.png)
+
+The Bucket page will have a lot of configurations you can set, but you can leave
+the settings in the default state if there’s nothing you need to customize. We
+have named this example bucket `updatedbikedata` as you can see below.
+
+![gcp_bucket_options.png](/uploads/images/2022-06-07/gcp_bucket_options.png)
+
+Now you can save your changes and you’ll be redirected to the `Bucket Details`
+page and you’ll see the bucket you just created.
+
+![created_gcp_bucket.png](/uploads/images/2022-06-07/created_gcp_bucket.png)
+
+### Get your credentials
+
+Since you have the bucket created, we need to get the credentials to connect the
```

Comment:
It might be worth to divide this section multiple subsections:
1. **Users who installed** `google-cloud-sdk`: run `gcloud auth application-default login` to authenticate to google cloud using the default user, they will have access to the bucket through dvc. Note that this case *does not require creation of a service account* unless some more complex scenarios are being considered (i.e. CI/CD).
2. **Users who installed the `google-cloud-sdk` but do want to use a service account:** create the service account, then run `gcloud auth activate-service-account SERVICE_ACCOUNT@DOMAIN.COM --key-file=/path/key.json`. Everything should work, requiring no extra configuration (although I have not tried), here are the [docs](https://cloud.google.com/sdk/gcloud/reference/auth/activate-service-account).
3. **Users who do not have `google-cloud-sdk` installed:** will have to follow through the service account guide to create a service account and then provide the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to provide the path to the service account credentials.

Reply:
This is a really good suggestion! I think this section is ok as it is because we're walking them through a pretty generic setup. But I will update the `Add GCP credentials` settings to highlight these different commands.